### PR TITLE
MTDSA-13507: TYS changes for Create and Amend a UK Property Business Annual Submission

### PR DESCRIPTION
**Pull request by GazalNebHMRC**

### CODE REVIEW

**Review by SamuelJacobsHMRC — CHANGES_REQUESTED** [review]

File: app/v2/connectors/AmendUkPropertyAnnualSubmissionConnector.scala [hunk 1]
```diff
@@ -32,11 +31,20 @@ class AmendUkPropertyAnnualSubmissionConnector @Inject()(val http: HttpClient, v
...
-    // Note that MTD tax year format is used
+    import request._
+
+    val downstreamUri =
+      if (taxYear.useTaxYearSpecificApi) {
+        TaxYearSpecificIfsUri[Unit](s"income-tax/business/property/annual/${taxYear.asTysDownstream}/${nino.nino}/$businessId")
+      } else {
+        // Note that MTD tax year format is used
+        IfsUri[Unit](
+          s"income-tax/business/property/annual?taxableEntityId=${nino.nino}&incomeSourceId=$businessId&taxYear=${request.taxYear.asMtd}")
```
[the remark this answers is not in the document]
Comment (GazalNebHMRC) [reply app/v2/connectors/AmendUkPropertyAnnualSubmissionConnector.scala]: Updated.

File: it/v2/endpoints/AmendUkPropertyAnnualSubmissionControllerISpec.scala [hunk 2]
```diff
@@ -161,27 +163,60 @@ class AmendUkPropertyAnnualSubmissionControllerISpec extends V2IntegrationBaseSp
        """.stripMargin
   }
 
+  private trait NonTysTest extends Test {
+    def taxYear: String                            = "2022-23"
+    def downstreamTaxYear: String                  = "2022-23"
+    def downstreamQueryParams: Map[String, String] = ifsQueryParams
```
[the remark this answers is not in the document]
Comment (GazalNebHMRC) [reply it/v2/endpoints/AmendUkPropertyAnnualSubmissionControllerISpec.scala]: Makes sense; updated.

File: app/v2/services/AmendUkPropertyAnnualSubmissionService.scala [hunk 3]
```diff
@@ -39,15 +39,13 @@ class AmendUkPropertyAnnualSubmissionService @Inject()(connector: AmendUkPropert
...
-      ifsResponseWrapper <- EitherT(connector.amendUkPropertyAnnualSubmission(request)).leftMap(mapDownstreamErrors(ifsErrorMap))
-    } yield ifsResponseWrapper
+    val result = EitherT(connector.amendUkPropertyAnnualSubmission(request)).leftMap(mapDownstreamErrors(downstreamErrorMap))
 
     result.value
   }
 
-  private def ifsErrorMap =
-    Map(
+  private def downstreamErrorMap = {
```
[the remark this answers is not in the document]
Comment (GazalNebHMRC) [reply app/v2/services/AmendUkPropertyAnnualSubmissionService.scala]: Updated.

File: test/v2/services/AmendUkPropertyAnnualSubmissionServiceSpec.scala [hunk 4]
```diff
@@ -39,6 +39,56 @@ class AmendUkPropertyAnnualSubmissionServiceSpec extends UnitSpec {
   val taxYear: TaxYear               = TaxYear.fromMtd("2020-21")
   implicit val correlationId: String = "X-123"
 
+  "service" when {
+    "service call successful" should {
+      "return mapped result" in new Test {
+        MockAmendUkPropertyAnnualSubmissionConnector
+          .amendUkProperty(request)
```
[the remark this answers is not in the document]
Comment (GazalNebHMRC) [reply test/v2/services/AmendUkPropertyAnnualSubmissionServiceSpec.scala:46]: Makes sense. I have updated now, thanks.

**Review by SamuelJacobsHMRC — APPROVED** [review]

**Pull request merged**